src/main/drivers/bus_i2c_stm32f10x.c

```diff
@@ -473,35 +473,38 @@ uint16_t i2cGetErrorCounter(void)
 static void i2cUnstick(IO_t scl, IO_t sda)
 {
     int i;
-    int timeout = 100;
 
     IOHi(scl);
     IOHi(sda);
 
     IOConfigGPIO(scl, IOCFG_OUT_OD);
     IOConfigGPIO(sda, IOCFG_OUT_OD);
 
-    for (i = 0; i < 8; i++) {
+    // Analog Devices AN-686
+    // We need 9 clock pulses + STOP condition
+    for (i = 0; i < 9; i++) {
```

Comment:
@SteveCEvans: The code is not copied unchanged -  you are incrementing the magic number, and you have added a comment (which is a good thing). The point that I am making is just that it's an even better approach to use a define, instead of using a magic number and then commenting on what it stands for. Also, using a define instead of the magic number will not propagate into `bus_i2c_hal.c` any more than your increment of the magic number does.

Reply:
@mikeller The i2cUnstick() routine is copied from bus_i2c_hal.c to bus_i2c_stm32f10x.c.

Reply:
Ah, now I see your point. Just means that the code in `bus_i2c_hal.c` isn't using best practices either. ;-)

Reply:
So, shall I make the requested changes to both? I've made the changes and I'm ready to push.

Reply:
Yes please.